From 7a05dd1874d3158683e3b10cc3dc4f09e1b2165f Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Fri, 29 Sep 2023 11:32:21 +0200
Subject: [PATCH 420/841] feat(stm32mp2): increase STM32MP_HW_CONFIG_MAX_SIZE

Increase STM32MP_HW_CONFIG_MAX_SIZE to allow usage of
bigger U-Boot DT.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e5d71a4e1da547e04ee77aabdcd422d25d5e3630 feat(stm32mp2): increase STM32MP_HW_CONFIG_MAX_SIZE]
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Change-Id: Ie5ea02ff8e50a384e4f8f327b14bc1f5a175a429
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/330508
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index a29df0454..b84627bcf 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -205,7 +205,7 @@ enum ddr_type {
 #define STM32MP_BL33_MAX_SIZE		U(0x400000)
 #define STM32MP_HW_CONFIG_BASE		(STM32MP_BL33_BASE + \
 					STM32MP_BL33_MAX_SIZE)
-#define STM32MP_HW_CONFIG_MAX_SIZE	U(0x20000)
+#define STM32MP_HW_CONFIG_MAX_SIZE	U(0x40000)
 #define STM32MP_SOC_FW_CONFIG_MAX_SIZE	U(0x10000) /* 64kB for BL31 DT */
 
 /* Define maximum page size for NAND devices */
-- 
2.47.2

